youtube_dl/extractor/torrins.py

```diff
@@ -0,0 +1,136 @@
+from __future__ import unicode_literals
+
+import re
+
+from .common import InfoExtractor
+from ..compat import (
+    compat_kwargs,
+)
+from ..utils import (
+    ExtractorError,
+    urlencode_postdata,
+)
+
+
+class TorrinsIE(InfoExtractor):
+    IE_NAME = 'torrins'
+    _VALID_URL = r'''(?x)
+                    https?://
+                        www\.torrins\.com/(?:guitar|piano|bass)-lessons/(?:song-lessons|style-genre)/[^/]+/(?P<id>[^/]+)/(?P<lesson_id>[^(\.)]+)\.html
+                    '''
+    _LOGIN_URL = 'https://www.torrins.com/services/user/sign-in'
+    _ORIGIN_URL = 'https://www.torrins.com'
+    _NETRC_MACHINE = 'torrins'
+
+    _TEST = {
+        'url': 'https://www.torrins.com/guitar-lessons/song-lessons/english-songs/another-brick-in-the-wall/song-demo.html',
+        'info_dict': {
+            'id': 'another-brick-in-the-wall',
+            'lesson_id': 'song-demo',
+            'ext': 'mp4',
+            'title': 'Another Brick in the Wall Guitar - Song Demo',
+            'description': 'md5:c0d51f6f21ef4ec65f091055a5eef876',
+            'duration': 579.29,
+        },
+        'skip': 'Requires torrins premium account credentials',
+    }
+
+    def _handle_error(self, response):
+        if not isinstance(response, dict):
+            return
+        error = response.get('error')
+        if error:
+            error_str = 'Torrins returned error #%s: %s' % (error.get('code'), error.get('message'))
+            error_data = error.get('data')
+            if error_data:
+                error_str += ' - %s' % error_data.get('formErrors')
+            raise ExtractorError(error_str, expected=True)
+
+    def _download_webpage(self, *args, **kwargs):
+        kwargs.setdefault('headers', {})['User-Agent'] = 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_5) AppleWebKit/603.2.4 (KHTML, like Gecko) Version/10.1.1 Safari/603.2.4'
```

Comment:
Why is that necessary?

Reply:
Torrins rejects request made from all non-browser user agents.

Reply:
youtube-dl does not use non-browser user agent.

Reply:
In that case, I will clean this up.